### PR TITLE
Saving state_dict of optimizer and lr_scheduler

### PR DESCRIPTION
**Pull request by zhu-han**

### CODE REVIEW

**Comment by danpovey** [comment]
OK.  Are you confident that this is all correct?  I don't have enough familiarity with PyTorch optimizers to know.

**Comment by danpovey** [comment]
.. thanks!!

**Comment by zhu-han** [comment]
Yes, I think this is correct. I checked the learning rate and loss curve, training from an intermediate epoch is consistent with training from start.

**Comment by danpovey** [comment]
Fantastic, thanks!  Merging.
